### PR TITLE
vtk.js PolyData browser pipeline support

### PR DESCRIPTION
**Pull request by thewtex**
@jourdain @floryst @agirault @jcfr

### CODE REVIEW

File: doc/content/api/node_pipelines.md [hunk 1]
```diff
@@ -21,12 +21,21 @@ Run an itk.js Emscripten module with Node.js.
...
-  <dt>`IOTypes.Image`</dt><dd>An [`itk/Image`](./Image.html).</dd>
-  <dt>`IOTypes.Mesh`</dt><dd>An [`itk/Mesh`](./Mesh.html).</dd>
+  <dt>`IOTypes.Image`</dt><dd>An [`itk/Image`](./Image.html). To generate this data type in C++, use the [`itk::ImageFileWriter`](https://itk.org/Doxygen/html/classitk_1_1ImageFileWriter.html), add `BridgeJavaScript` module to the ITK `find_package` `COMPONENTS` in your *CMakeLists.txt*, and use `.json` for the image file name extension.</dd>
+  <dt>`IOTypes.Mesh`</dt><dd>An [`itk/Mesh`](./Mesh.html). To generate this data type in C++, use the [`itk::MeshFileWriter`](https://itk.org/Doxygen/html/classitk_1_1MeshFileWriter.html), add `BridgeJavaScript` module to the ITK `find_package` `COMPONENTS` in your *CMakeLists.txt*, and use `.json` for the image file name extension.</dd>
+  <dt>`IOTypes.vtkPolyData`</dt><dd>A vtk.js [`PolyData`](https://kitware.github.io/vtk-js/docs/structures_PolyData.html). To generate this data type in C++, use the [`vtkJSONDataSetWriter`](https://vtk.org/doc/nightly/html/classvtkJSONDataSetWriter.html), add `vtkIOExport` module to the VTK `find_package` `COMPONENTS` in your *CMakeLists.txt*</dd>
```
Comment (jourdain) [inline doc/content/api/node_pipelines.md]: The vtkJSONDataSetWriter will write polydata as well as imagedata to a vtk.js bundle. I'm not sure to understand what you mean by "to generate that data type use ...".

Comment (jourdain) [reply doc/content/api/node_pipelines.md]: The vtkXMLPolyDataWriter which generate vtp files would be more appropriate?

Comment (jourdain) [reply doc/content/api/node_pipelines.md]: I guess reading further, it seems that the vtk.js format might be indeed a better fit for a format of exchange between Emscripten and plain JS.

Comment (thewtex) [reply doc/content/api/node_pipelines.md]: Yes, that's it: the vtkJSONDataSetWriter format enables efficient  / zero-copy / zero-computation conversions from the Emscripten WebAssembly virtual filesystem.

Similarly, we could add vtk.js vtkImageData support if there is a need for it.

Comment (jourdain) [reply doc/content/api/node_pipelines.md]: ok np.

File: doc/content/api/node_pipelines.md [hunk 2]
```diff
@@ -21,12 +21,21 @@ Run an itk.js Emscripten module with Node.js.
 <dl>
   <dt>`IOTypes.Text`</dt><dd>A UTF8-encoded string.</dd>
   <dt>`IOTypes.Binary`</dt><dd>A binary [`Uint8Array`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array).</dd>
-  <dt>`IOTypes.Image`</dt><dd>An [`itk/Image`](./Image.html).</dd>
-  <dt>`IOTypes.Mesh`</dt><dd>An [`itk/Mesh`](./Mesh.html).</dd>
+  <dt>`IOTypes.Image`</dt><dd>An [`itk/Image`](./Image.html). To generate this data type in C++, use the [`itk::ImageFileWriter`](https://itk.org/Doxygen/html/classitk_1_1ImageFileWriter.html), add `BridgeJavaScript` module to the ITK `find_package` `COMPONENTS` in your *CMakeLists.txt*, and use `.json` for the image file name extension.</dd>
```
Comment (agirault) [inline doc/content/api/node_pipelines.md]: The `inline code formatting` (with back-ticks around) does not seem to play weel with the `<dl><dt>` html tags, see picture below:

screen shot 2019-02-06 at 11 21 49 am

Comment (agirault) [reply doc/content/api/node_pipelines.md]: It seems the issue applies to the hyperlinks also.

Comment (thewtex) [reply doc/content/api/node_pipelines.md]: Interesting -- I do not observe this locally (markdown parser version issue?) but it does ring a bell from the past -- correcting along with the `the the` issue.

**Comment by thewtex** [comment]
Thanks for the reviews :+1:

File: doc/content/api/node_pipelines.md (hunk 2 above)
Comment (agirault) [reply doc/content/api/node_pipelines.md]: Doesn't seem resolved, was it?
https://github.com/thewtex/itk-js/blob/48c44b44415711cd8de46f06e772dc22f97e03b5/doc/content/api/node_pipelines.md

**Pull request merged**